extensions/wikia/PortableInfobox/tests/PortableInfoboxRenderServiceHelperTest.php

```diff
@@ -296,4 +301,127 @@ public function testIsTypeSupportedInTemplatesDataProvider() {
 			]
 		];
 	}
+
+	/**
+	 * @param $mockParams
+	 * @param $isWikiaMobile
+	 * @param $result
+	 * @param $description
+	 * @dataProvider testGetAdjustedImageSizeDataProvider
+	 */
+	public function testGetAdjustedImageSize( $mockParams, $isWikiaMobile, $result, $description ) {
+		$mock = $this->getMockBuilder( 'Wikia\PortableInfobox\Helpers\PortableInfoboxRenderServiceHelper' )
+			->setMethods( [ 'isWikiaMobile' ] )
+			->getMock();
+		$mock->expects( $this->any() )->method( 'isWikiaMobile' )->will( $this->returnValue( $isWikiaMobile ) );
+
+		$file = $this->createWikiaFileHelperMock( $mockParams );
+
+		$this->assertEquals(
+			$result,
+			$mock->getAdjustedImageSize( $file ),
+			$description
+		);
+	}
+
+	public function testGetAdjustedImageSizeDataProvider() {
+		return [
+			[
+				'mockParams' => [
+					'fileHeight' => 2000,
+					'fileWidth' => 3000
+				],
+				'isWikiaMobile' => false,
+				'result' => [
+					'height' => 500,
+					'width' => 270
```

Comment:
this are the sizes we pass to `transform` function, not the sizes we want image to have. It automatically adjusts the correct sizes, that is creates thumbnail with sizes not bigger than passed, keeping the aspect ratio. That's why, the real sizes of the image (which we pass to html attrs) we take from:
`$data[ 'height' ] = $thumbnail->getHeight();
$data[ 'width' ] = $thumbnail->getWidth();`